lib/DDG/Spice/CodeDay.pm

```diff
@@ -0,0 +1,56 @@
+package DDG::Spice::CodeDay;
+# ABSTRACT: Get all CodeDay events, sorted by closest to farthest
+
+use DDG::Spice;
+use Text::Trim;
+
+spice is_cached => 0;
+
+name "CodeDay";
+source "CodeDay.org";
+icon_url "https://codeday.org/assets/img/favicon.ico";
+description "Get information about a CodeDay or find the nearest one.";
+primary_example_queries "codeday near me", "when is codeday?", "codeday san diego";
+secondary_example_queries "codeday", "codeday seattle", "what is codeday bay area?";
+category "programming";
+topics "programming", "geek", "computing", "entertainment";
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/CodeDay.pm";
+attribution github => ["tjhorner", "TJ Horner"],
+            twitter => "tjhorner";
+
+spice to => 'https://clear.codeday.org/api/regions/$1?term=$2&lat=$3&lng=$4&limit=50';
```

Comment:
@tjhorner sorry to clarify I mean we could store a mapping of state names to lat/long here. So then we could stick to only using the one API endpoint. That way `codeday near me` would use lat long from `$loc` and `codeday san diego` would lookup the lat/long in our local textfile, and use that with the search endpoint. This way we don't need to fiddle with using two different API endpoints, or returning blank parameters.

The third solution is to adjust your API and this code so that we can always just return an array of defined data, and don't need to worry about passing a blank space as a value.

e.g.

```perl
spice from => "(.+)/(.+);
spice to => "https://clear.codeday.org/api/lookup?type=$1&term=$2

...

handle query_lc => sub {
    ...
    return "region", $city_name if isCitySearch;
    return "latlong", "$lat|$long" if isLocalSearch;
```

This would require the API to handle things slightly differently, but enables our `return` to be normalized a little and then we don't need to worry about NGINX mishandling empty array values.

What do you think?